bus-mapping/src/circuit_input_builder.rs

```diff
@@ -572,4 +595,21 @@ impl<P: JsonRpcClient> BuilderClient<P> {
         )?;
         Ok((builder, eth_block))
     }
+
+    /// Check if a tx is invalid or not.
+    pub fn check_invalid_txs(
+        &self,
+        block: &eth_types::Block<eth_types::Transaction>,
+        geth_traces: &mut [eth_types::GethExecTrace],
+        txs_rlp: &eth_types::Bytes,
+    ) {
+        let txs:Vec<eth_types::Transaction> = Rlp::new(txs_rlp).as_list().expect("invalid txs rlp");
+        // Skip the 1st tx which is an anchor tx
+        if block.transactions.len() == 0 && geth_traces.len() == 0 {
+            return
+        }
+        for (tx, geth_trace) in block.transactions[1..].iter().zip(geth_traces[1..].iter_mut()) {
```

Comment:
Is there a reason to skip the anchor transaction here? In theory could of course be marked as invalid but then elsewhere in the anchor specific logic we would just require it to be valid or something like that?

Reply:
I think we will/should have an anchor circuit just for anchor tx, which will be very different. So here, we can skip anchor tx.